web-registry/src/components/molecules/DenomIcon/DenomIcon.tsx

```diff
@@ -0,0 +1,31 @@
+import { Box, SxProps, Theme } from '@mui/material';
+import {
+  AXELAR_USDC_DENOM,
+  EEUR_DENOM,
+  GRAVITY_USDC_DENOM,
+  REGEN_DENOM,
+} from 'config/allowedDenoms';
+
+import AxlUsdcIcon from 'web-components/lib/components/icons/coins/AxlUsdcIcon';
+import EeurIcon from 'web-components/lib/components/icons/coins/EeurIcon';
+import GravUsdcIcon from 'web-components/lib/components/icons/coins/GravUsdcIcon';
+import { RegenTokenIcon } from 'web-components/lib/components/icons/RegenTokenIcon';
+
+export interface Props {
+  denom: string;
+  sx?: SxProps<Theme>;
+  iconSx?: SxProps<Theme>;
+}
+
+const DenomIcon = ({ denom, sx = [], iconSx = [] }: Props): JSX.Element => {
```

Comment:
This component can be reused wherever we need to display a coin logo based on denom.